src/node.rs

```diff
@@ -91,23 +95,16 @@ impl Node {
                 println!("{}", command.parse()?);
                 Ok(())
             }
-            None => match (
-                self.network,
-                self.miner.is_some(),
-                self.trial,
-                self.sync,
-                self.pool_address.is_some(),
-                self.mining_pool,
-            ) {
-                (2, _, _, true, false, false) => self.start_server::<Testnet2, SyncNode<Testnet2>>().await,
-                (2, true, false, false, false, false) => self.start_server::<Testnet2, Miner<Testnet2>>().await,
-                (2, false, false, false, false, false) => self.start_server::<Testnet2, Client<Testnet2>>().await,
-                (2, true, false, false, false, true) => self.start_server::<Testnet2, Operator<Testnet2>>().await,
-                (2, true, false, false, true, false) => self.start_server::<Testnet2, Prover<Testnet2>>().await,
-                (2, true, true, false, false, false) => self.start_server::<Testnet2, MinerTrial<Testnet2>>().await,
-                (2, false, true, false, false, false) => self.start_server::<Testnet2, ClientTrial<Testnet2>>().await,
-                (2, true, true, false, false, true) => self.start_server::<Testnet2, OperatorTrial<Testnet2>>().await,
-                (2, true, true, false, true, false) => self.start_server::<Testnet2, ProverTrial<Testnet2>>().await,
+            None => match (self.network, &self.miner, &self.operator, &self.prover, self.trial, self.sync) {
```

Comment:
at this point it might be a good idea to introduce a helper `struct` to hold these init values

Reply:
I've made an issue out of this, and left it out of the scope of this PR for now.